client/service-container-chain/src/spawner.rs

```diff
@@ -256,16 +256,19 @@ async fn try_spawn<SelectSyncMode: TSelectSyncMode>(
     if !start_collation {
         collation_params = None;
 
-        log::info!("This is a syncing container chain, using random ports");
+        log::info!("This is a syncing container chain, using random ports unless they are explicitly provided as CLI args");
         // Use random ports to avoid conflicts with the other running container chain
+        // Don't override provided port.
+        // TODO: How does this prevent conflicts? All containers will have the same ports.
```

Comment:
It's because in this branch we are spawning a container chain that will only be alive for 1 session, and will restart later with the correct ports. So we only need the ports to not conflict with the CLI ports, because that can cause conflict with the other chain that is running. At first I used different ports every time (`rng.gen()`) but then I realized that a fixed port will never conflict while a random port may conflict sometimes.

If you need to run a container chain for more than 1 session, you shouldn't enter this branch, the default is to use CLI ports.

Reply:
*fixed*